### PR TITLE
Kernels for GroupNorm

### PR DESCRIPTION
**Pull request by pramodith**
## Summary
 Implementation of group norm that achieves output parity with torch's `GroupNorm`. 

This is feature is a part of #285 

## Details
The formulas/equations involved in `GroupNorm` are the same as `LayerNorm/BatchNorm`. The main differences lie in the axis along which the mean and std are computed + the dimensions of the Affine transformation parameters.

In group norm W and B are of shape `(n_channels)`, however the mean and std are calculated over all the channels in a given group.

## Testing Done

Testing was done on a A100 PCIE and a A100 SXM-4.

We see an increase in speed, while the total memory used remains about the same. Note that benchmarking was done using a batch size of 128, Hidden Dim size of 512 and the number of channels per group fixed at 4.

These results look very similar to the layer norm benchmark too.

group_norm_memory

group_norm_speed

- Hardware Type: <BLANK>
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [X] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by lancerts** [comment]
Very solid PR!

**Review by lancerts — APPROVED** [review]

**Review by lancerts — APPROVED** [review]

File: src/liger_kernel/ops/group_norm.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            x_hat = (X - mean) * rstd
+            dW += tl.sum(UPSTREAM_grad * x_hat)
+            dB += tl.sum(UPSTREAM_grad)
+
+            wdy = W * UPSTREAM_grad
+            c1 += tl.sum(x_hat * wdy)
+            c2 += tl.sum(wdy)
+
+        # Need to ensure additions to the same channel are atomic
+        tl.atomic_add(DW_ptr + channel_idx, dW.to(dtype))
```
[the remark this answers is not in the document]
Comment (ByronHsu) [reply src/liger_kernel/ops/group_norm.py:177]: what kind of testing? run on a 4 GPUs env to ensure the kernel working fine on a single GPU? Not sure how this is related to multi gpu. My understanding is the kernel only happens on 1 gpu

**Comment by ByronHsu** [comment]
@pramodith can you update the readme to include groupnorm

**Comment by pramodith** [comment]
> @pramodith can you update the readme to include groupnorm

Will do tomorrow!

**Pull request merged**